### PR TITLE
`chain-spec-builder`: cleanup

### PR DESCRIPTION
**Pull request by michalkucharczyk**
This PR removes:
-  `New`, `Generate`, `Edit` commands,
- `kitchensink` dependency
from the `chain-spec-builder` util.

New `convert-to-raw`, `update-code` commands were added.

Additionally renames the `runtime` command (which was added in #1256)  to `create`.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4225644

File: substrate/bin/utils/chain-spec-builder/bin/main.rs [hunk 1]
```diff
@@ -17,83 +17,24 @@
...
-			authority_seeds,
-			nominator_accounts,
-			endowed_accounts,
-			sudo_account,
-		}) =>
-			generate_chain_spec(authority_seeds, nominator_accounts, endowed_accounts, sudo_account),
-		ChainSpecBuilderCmd::Runtime(cmd) => generate_chain_spec_for_runtime(&cmd),
 		ChainSpecBuilderCmd::Edit(EditCmd {
 			ref input_chain_spec,
 			ref runtime_wasm_path,
```
Comment (skunert) [inline substrate/bin/utils/chain-spec-builder/bin/main.rs:40]: Nit: Should any of these be mandatory? Just doing
chain-spec-builder edit --input-chain-spec chain_spec.json does silently nothing.

File: substrate/bin/utils/chain-spec-builder/src/lib.rs [hunk 2]
```diff
@@ -171,153 +170,32 @@ struct DefaultCmd {
...
 }
 
-/// Verifies provided input chain spec. If the runtime is provided verification is performed against
+/// Verifies provided input chain spec.
 /// new runtime.
 #[derive(Parser, Debug, Clone)]
 pub struct VerifyCmd {
-	/// Chain spec to be edited
+	/// Chain spec to be edited.
 	#[arg(long, short)]
```
Comment (skunert) [inline substrate/bin/utils/chain-spec-builder/src/lib.rs:189]:
```suggestion
```
What do you think about removing it? Then we could to `chain-spec-builder verify chain_spec.json` without the need to always specify the long commands. I think there are also some other commands that take only one command but still require knowledge of a command line flag. But its only a nit, up to you.

**Review by skunert — APPROVED** [review]

File: substrate/bin/utils/chain-spec-builder/src/lib.rs [hunk 3]
```diff
@@ -171,153 +170,32 @@ struct DefaultCmd {
 /// can be updated with the runtime provided in the command line.
 #[derive(Parser, Debug, Clone)]
 pub struct EditCmd {
-	/// Chain spec to be edited
+	/// Chain spec to be edited.
 	#[arg(long, short)]
```
Comment (bkchr) [inline substrate/bin/utils/chain-spec-builder/src/lib.rs:174]: As @skunert said, this should be some required argument.

File: substrate/bin/utils/chain-spec-builder/src/lib.rs [hunk 4]
```diff
@@ -21,31 +21,76 @@
...
+//! 	```
+//!
+//!		_Note_: No runtime API is called.
+//!
+//! - **Generate human readable chain spec using provided full genesis config.**
+//!
+//!   ```text
+//!   chain-spec-builder create -r runtime.wasm full -c full-genesis-config.json
+//!   ```
+//!   _Note_: No runtime API is called.
```
Comment (bkchr) [inline substrate/bin/utils/chain-spec-builder/src/lib.rs:74]:
```suggestion
```

File: substrate/bin/utils/chain-spec-builder/src/lib.rs [hunk 5]
```diff
@@ -21,31 +21,76 @@
...
+//! 	```text
+//! 	chain-spec-builder create -s -r runtime.wasm full -c full-genesis-config.json
+//! 	```
+//!
+//! 	_Note_: `GenesisBuilder::build_config` runtime function is called.
+//!
+//! - **Generate human readable chain spec using provided genesis config patch.**
+//!
+//!		```text
+//! 	chain-spec-builder create -r runtime.wasm patch -p patch.json
```
Comment (bkchr) [inline substrate/bin/utils/chain-spec-builder/src/lib.rs:64]:
```suggestion
//!	```text
//! 	chain-spec-builder create -r runtime.wasm patch -p patch.json
```

File: substrate/bin/utils/chain-spec-builder/src/lib.rs [hunk 6]
```diff
@@ -21,31 +21,76 @@
...
+//! 	```
+//!
+//! 	_Note:_ `GenesisBuilder::create_default_config` runtime function is called.
+//!
+//!
+//! - **Generate raw storage chain spec using genesis config patch.**
+//!
+//!		Patch the runtime's default genesis config with provided `patch.json` and generate raw
+//! storage (`-s`) version of chain spec:
+//!		```text
```
Comment (bkchr) [inline substrate/bin/utils/chain-spec-builder/src/lib.rs:46]:
```suggestion
//!	```text
//! 	chain-spec-builder create -r runtime.wasm default -d /dev/stdout
//! 	```
//!
//! 	_Note:_ `GenesisBuilder::create_default_config` runtime function is called.
//!
//!
//! - **Generate raw storage chain spec using genesis config patch.**
//!
//!		Patch the runtime's default genesis config with provided `patch.json` and generate raw
//! storage (`-s`) version of chain spec:
//!	```text
```

Comment (bkchr) [reply substrate/bin/utils/chain-spec-builder/src/lib.rs:46]: And please make the `GenesisBuilder::create_default_config` references, intra doc links. [`GenesisBuilder::create_default_config`](sp_genesis_builder::GenesisBuilder::create_default_config)

**Review by bkchr — APPROVED** [review]

File: substrate/bin/utils/chain-spec-builder/src/lib.rs (hunk 2 above)
Comment (michalkucharczyk) [reply substrate/bin/utils/chain-spec-builder/src/lib.rs:187]: make sense, fixed.

File: substrate/bin/utils/chain-spec-builder/bin/main.rs (hunk 1 above)
Comment (michalkucharczyk) [reply substrate/bin/utils/chain-spec-builder/bin/main.rs:40]: Seems that two separate commands would be better. How about:
- `update-code`
- `convert-to-raw`
?

Probably that would be better then one `edit` with unclear options.

File: substrate/bin/utils/chain-spec-builder/src/lib.rs (hunk 3 above)
Comment (michalkucharczyk) [reply substrate/bin/utils/chain-spec-builder/src/lib.rs:168]: https://github.com/paritytech/polkadot-sdk/pull/2174#discussion_r1399779900

**Comment by michalkucharczyk** [comment]
bot fmt

**Comment by michalkucharczyk** [comment]
bot fmt

File: substrate/bin/utils/chain-spec-builder/bin/main.rs (hunk 1 above)
Comment (michalkucharczyk) [reply substrate/bin/utils/chain-spec-builder/bin/main.rs:41]: done in: ed84b6b

File: substrate/bin/utils/chain-spec-builder/src/lib.rs (hunk 3 above)
Comment (michalkucharczyk) [reply substrate/bin/utils/chain-spec-builder/src/lib.rs]: done in: ed84b6b

**Review by skunert — APPROVED** [review]

**Pull request merged**